src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
+  local ITEM_FILE="${BASEDIR}/${item}_${LOG_SUFFIX}.log"
+
+  if [[ ${SNAPSHOT} == "yes" ]]; then
+    ITEM_FILE="${SNAPSHOTSDIR}/${item}.log_snapshot_${DATE}"
+  fi
+
+  create_output_file "${ITEM_FILE}"
+  check_output_file "${ITEM_FILE}"
+  print_netstat "${ITEM_FILE}"
+
+  # check to see if optional netstat summary report should be run
+  if [[ ! -z ${USENETSTATSUM} ]]; then
+    if [[ ${USENETSTATSUM,,} == "yes" ]]; then
+      print_blankline "${ITEM_FILE}"
+      print_netstat_sum "${ITEM_FILE}"
+    fi
+  fi
 }
 
 # manage mysql report
 run_mysql_report() {
-	if [ $SNAPSHOT = "yes" ]
-	then
-		MYSQL_FILE="$SNAPSHOTSDIR/mysql.log_snapshot_$DATE";
-	fi
-
-	create_output_file $MYSQL_FILE
-	check_output_file $MYSQL_FILE
-	print_mysql
-
-	# check to see if the optional mysql process list should be generated
-	if [ ! -z $USEMYSQLPROCESSLIST ]
-	then
-		if [ $USEMYSQLPROCESSLIST = "yes" -o $USEMYSQLPROCESSLIST = "YES" ]
-		then
-			print_blankline $MYSQL_FILE
-			print_mysql_procs
-		fi
-	fi
-
-	if [ $USEINNODB = "yes" -o $USEINNODB = "YES" ]
-	then
-		# send df -h output to output file
-		print_blankline $MYSQL_FILE
-		print_mysql_innodb_status
-	fi
+  local item="mysql"
+  local ITEM_FILE="${BASEDIR}/${item}_${LOG_SUFFIX}.log"
+
+  if [[ ${SNAPSHOT} == "yes" ]]; then
+    ITEM_FILE="${SNAPSHOTSDIR}/${item}.log_snapshot_${DATE}"
+  fi
+
+  create_output_file "${ITEM_FILE}"
+  check_output_file "${ITEM_FILE}"
+  print_mysql "${ITEM_FILE}"
+
+  # check to see if the optional mysql process list should be generated
+  if [[ ! -z ${USEMYSQLPROCESSLIST} ]]; then
+    if [[ ${USEMYSQLPROCESSLIST,,} == "yes" ]]; then
+      print_blankline "${ITEM_FILE}"
+      print_mysql_procs "${ITEM_FILE}"
+    fi
+  fi
+  if [[ ${USEINNODB,,} == "yes" ]]; then
+    # send df -h output to output file
+    print_blankline "${ITEM_FILE}"
+    print_mysql_innodb_status "${ITEM_FILE}"
+  fi
 }
 
-# copy the last log file set to $BACKUPDIR
+# copy the last log file set to ${BACKUPDIR}
 create_backup() {
-	for f in $BACKUP_ITEMS; do
-		ls -1t ${BASEDIR}/${f}_*.log | \
-		grep ".*_[0-9]\{8\}-[0-9]\{6\}.log" | \
-		head -1 | \
-		xargs -I {} cp {} $BACKUPDIR
-	done
+  for f in ${BACKUP_ITEMS}; do
+    ls -1t ${BASEDIR}/${f}_*.log | \
+    grep -m1 ".*_[0-9]\{8\}-[0-9]\{6\}.log" | \
+    xargs -I {} cp {} ${BACKUPDIR}
+  done
 }
 
 # check enough disk space is free
 check_disk_space() {
-        if [[ $MIN_FREE_SPACE -eq 0 ]]
-		then
-        	return 0
-        else
-        	FREE_SPACE=$(df -BM $BASEDIR|grep -v 1M-blocks|awk '{print $4}')
-        	FREE_SPACE=${FREE_SPACE%M}
-        	if [[ $MIN_FREE_SPACE -ge $FREE_SPACE ]]
-			then
-        		wall "Unable to run recap due to not enough disk space"
-        	fi
-        fi
+  if [[ ${MIN_FREE_SPACE} -eq 0 ]]; then
+    return 0
+  else
+    FREE_SPACE=$( df -BM ${BASEDIR} | grep -v 1M-blocks | awk '{print $4}' )
+    FREE_SPACE=${FREE_SPACE%M}
+    if [[ ${MIN_FREE_SPACE} -ge ${FREE_SPACE} ]]; then
```

Comment:
Gonna need quotes here, because "check_disk_space" runs prior to "create_output_dirs", so it's possible you just ran 'df' on a directory that doesn't exist, which would cause ${FREE_SPACE} to be null.